runtime/lua/vim/lsp/handlers.lua

```diff
@@ -476,15 +476,17 @@ function M.signature_help(_, result, ctx, config)
     vim.tbl_get(client.server_capabilities, 'signatureHelpProvider', 'triggerCharacters')
   local ft = vim.bo[ctx.bufnr].filetype
   local lines, hl = util.convert_signature_help_to_markdown_lines(result, ft, triggers)
-  if vim.tbl_isempty(lines) then
+  if not lines or vim.tbl_isempty(lines) then
```

Comment:
Wouldn't it be better to normalize the markdown lines in `convert_signature_help_to_markdown_lines` ?

Reply:
Not sure I understand what you mean by "normalize".

Reply:
I mean: make the return of `convert_signature_help_to_markdown_lines` always return the signature in the first line, without ```` ``` ````s.
That function naming looks like a good candidate for that behavior

(note that I'm an outsider here, I don't have much knowledge on this part of the code, I'm just curious why you made a check after a conversion function instead of changing the conversion function to always return the same format of output)

Reply:
`convert_signature_help_to_markdown_lines` converts the input text to Markdown content. Given that we want to display the signature with the Treesitter highlights of the respective language and we use the Markdown parser for these floating windows, the code block is necessary.

Reply:
The reason why this is conditional is because LSP Markdown content could also be `plaintext`, and in such case we don't add any Markdown decorations and display the content as is.

Reply:
Make sense, thanks 👍
(can mark this discussion as resolved)

Reply:
It's okay! I appreciate your input and please ask all the questions you want :)